### PR TITLE
shellIntegration.ps1: escape values in "E" (executed command) and "P" (property KV) codes

### PR DESCRIPTION
**Pull request by rwe**
This implements the string escaping scheme documented for the "E" (executed) and "P" (property KV) codes. This is "pure" `zsh` and relies on no external utilities.

Although currently the only "required" escapes are backslash, semicolon, `\a`, and perhaps newlines, this conservatively escapes all non-alphanumeric characters.

Backslashes are doubled, non-alphanumerics are hex-escaped.

Sibling to:
* #165631
* #165632
* #165633
* #165635

Relates to or resolves parts of:
* #157546
* #155639

### Caveat
Unaddressed here, but more plain to see than in the other shells, is that the encoding of multibyte codepoints is not currently well-defined by the escaping scheme. This chooses to encode as UTF-8 in order to make them byte-representable at all. Regardless, `shellIntegrationAddon.ts` severely mishandles the interpretation of escapes anyway, even in the ASCII range, so the point is slightly moot.

### CODE REVIEW

**Comment by Tyriar** [comment]
I'm seeing this when running recent command when I fix conflicts and test:

image

**Comment by Tyriar** [comment]
For this:

image

Here's the string we get:
image

**Comment by Tyriar** [comment]
image

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1 [hunk 1]
```diff
@@ -14,12 +14,23 @@ if ($ExecutionContext.SessionState.LanguageMode -ne "FullLanguage") {
...
+	[regex]::Replace($value, '[^a-zA-Z0-9]', { param($match)
+		# Backslashes must be doubled.
+		if ($match.Value -eq '\') {
+			'\\'
+		} else {
+			# Any other characters are encoded as their UTF-8 hex values.
+			-Join (
+				[System.Text.Encoding]::UTF8.GetBytes($match.Value)
+				| ForEach-Object { param($byte) '\x{0:x2}' -f $byte }
+			)
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1:31]: Any ideas on the problem in the screenshots @rwe? This is testing my pwsh knowledge limits 🙂

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1 (hunk 1 above)
Comment (rwe) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1:31]: Looks like I dumbed something in that last commit by naming the parameter…PowerShell isn't my forte either! Thanks for checking.

This gives the expected encoding, I'll push it up in a sec:

```diff
-				| ForEach-Object { param($byte) '\x{0:x2}' -f $byte }
+				| ForEach-Object { '\x{0:x2}' -f $_ }
```

Also note that without #165635, the editor itself incorrectly decodes messages, as described there.

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1:31]: Nice 👍. Yep, hoping to get to that one today. Thanks again for all the effort!

Comment (rwe) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1:31]: Ok, I hope rebasing on `main` is OK — that line is the only change from the merge you'd resolved. (Modulo a single blank line). Pushing up now.

**Review by Tyriar — APPROVED** [review]
Works now 👍

**Review by connor4312 — APPROVED** [review]

**Pull request merged**